tools/db_bench_tool.cc

```diff
@@ -228,6 +225,130 @@ DEFINE_string(
+    "\treadreverse   -- read N times in reverse order\n"
+    "\treadrandom    -- read N times in random order\n"
+    "\treadmissing   -- read N missing keys in random order\n"
+    "\treadwhilewriting      -- 1 writer, N threads doing random "
+    "reads\n"
+    "\treadwhilemerging      -- 1 merger, N threads doing random "
+    "reads\n"
+    "\treadwhilescanning     -- 1 thread doing full table scan, "
+    "N threads doing random reads\n"
+    "\treadrandomwriterandom -- N threads doing random-read, "
+    "random-write\n"
+    "\tupdaterandom  -- N threads doing read-modify-write for random "
+    "keys\n"
+    "\txorupdaterandom  -- N threads doing read-XOR-write for "
+    "random keys\n"
+    "\tappendrandom  -- N threads doing read-modify-write with "
+    "growing values\n"
+    "\tmergerandom   -- same as updaterandom/appendrandom using merge"
+    " operator. "
+    "Must be used with merge_operator\n"
+    "\treadrandommergerandom -- perform N random read-or-merge "
+    "operations. Must be used with merge_operator\n"
+    "\tnewiterator   -- repeated iterator creation\n"
+    "\tseekrandom    -- N random seeks, call Next seek_nexts times "
+    "per seek\n"
+    "\tseekrandomwhilewriting -- seekrandom and 1 thread doing "
+    "overwrite\n"
+    "\tseekrandomwhilemerging -- seekrandom and 1 thread doing "
+    "merge\n"
+    "\tcrc32c        -- repeated crc32c of 4K of data\n"
+    "\txxhash        -- repeated xxHash of 4K of data\n"
+    "\tacquireload   -- load N*1000 times\n"
+    "\tfillseekseq   -- write N values in sequential key, then read "
+    "them by seeking to each key\n"
+    "\trandomtransaction     -- execute N random transactions and "
+    "verify correctness\n"
+    "\trandomreplacekeys     -- randomly replaces N keys by deleting "
+    "the old version and putting the new version\n\n"
+    "\ttimeseries            -- 1 writer generates time series data "
+    "and multiple readers doing random reads on id\n\n"
+    "Meta operations:\n"
+    "\tcompact     -- Compact the entire DB; If multiple, randomly choose one\n"
+    "\tcompactall  -- Compact the entire DB\n"
+    "\tflush - flush the memtable\n"
+    "\tstats       -- Print DB stats\n"
+    "\tresetstats  -- Reset DB stats\n"
+    "\tlevelstats  -- Print the number of files and bytes per level\n"
+    "\tmemstats  -- Print memtable stats\n"
+    "\tsstables    -- Print sstable info\n"
+    "\theapprofile -- Dump a heap profile (if supported by this port)\n"
+    "\treplay      -- replay the trace file specified with trace_file\n"
+    "\tgetmergeoperands -- Insert lots of merge records which are a list of "
+    "sorted ints for a key and then compare performance of lookup for another "
+    "key "
+    "by doing a Get followed by binary searching in the large sorted list vs "
+    "doing a GetMergeOperands and binary searching in the operands which are"
+    "sorted sub-lists. The MergeOperator used is sortlist.h\n");
+#endif
```

Comment:
Can you explain the reason behind this change? This appears to make the code harder to maintain in the future.

Reply:
`DEFINE_string` is a macro and MSVC 14.0 reports a compiler error if we put #ifdef/#ifndef inside #define. The error looks like below:
```
##[error]tools\db_bench_tool.cc(226,0): Error C2121: '#': invalid character: possibly the result of a macro expansion
   223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2121: '#': invalid character: possibly the result of a macro expansion [D:\agent\_work\10\s\build\db_bench.vcxproj]
##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ')' before identifier 'ifndef'
   223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ')' before identifier 'ifndef' [D:\agent\_work\10\s\build\db_bench.vcxproj]
##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ';' before identifier 'ifndef'
   223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ';' before identifier 'ifndef' [D:\agent\_work\10\s\build\db_bench.vcxproj]
##[error]tools\db_bench_tool.cc(226,0): Error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
   223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [D:\agent\_work\10\s\build\db_bench.vcxproj]
##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ';' before identifier 'ROCKSDB_LITE'
   223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ';' before identifier 'ROCKSDB_LITE' [D:\agent\_work\10\s\build\db_bench.vcxproj]
##[error]tools\db_bench_tool.cc(226,0): Error C2059: syntax error: ')'
```
I searched the same issue on stackoverflow [ifdef-inside-define](https://stackoverflow.com/questions/5586429/ifdef-inside-define), [ifdef-inside-a-macro](https://stackoverflow.com/questions/7246512/ifdef-inside-a-macro), and it looks like a general rule but I am not sure why other compilers do not complain. Do you have better idea to make both the code easy to maintain and MSVC compiler happy?

Reply:
> `DEFINE_string` is a macro and MSVC 14.0 reports a compiler error if we put #ifdef/#ifndef inside #define. The error looks like below:
> 
> ```
> ##[error]tools\db_bench_tool.cc(226,0): Error C2121: '#': invalid character: possibly the result of a macro expansion
>    223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2121: '#': invalid character: possibly the result of a macro expansion [D:\agent\_work\10\s\build\db_bench.vcxproj]
> ##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ')' before identifier 'ifndef'
>    223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ')' before identifier 'ifndef' [D:\agent\_work\10\s\build\db_bench.vcxproj]
> ##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ';' before identifier 'ifndef'
>    223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ';' before identifier 'ifndef' [D:\agent\_work\10\s\build\db_bench.vcxproj]
> ##[error]tools\db_bench_tool.cc(226,0): Error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
>    223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [D:\agent\_work\10\s\build\db_bench.vcxproj]
> ##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ';' before identifier 'ROCKSDB_LITE'
>    223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ';' before identifier 'ROCKSDB_LITE' [D:\agent\_work\10\s\build\db_bench.vcxproj]
> ##[error]tools\db_bench_tool.cc(226,0): Error C2059: syntax error: ')'
> ```
> 
> I searched the same issue on stackoverflow [ifdef-inside-define](https://stackoverflow.com/questions/5586429/ifdef-inside-define), [ifdef-inside-a-macro](https://stackoverflow.com/questions/7246512/ifdef-inside-a-macro), and it looks like a general rule but I am not sure why other compilers do not complain. Do you have better idea to make both the code easy to maintain and MSVC compiler happy?

It's OK if there is no alternative.